src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/FindBoundaryCells.cpp

```diff
@@ -0,0 +1,130 @@
+#include "FindBoundaryCells.hpp"
+
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/Geometry/ImageGeom.hpp"
+
+using namespace complex;
+
+// -----------------------------------------------------------------------------
+FindBoundaryCells::FindBoundaryCells(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel, FindBoundaryCellsInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_InputValues(inputValues)
+, m_ShouldCancel(shouldCancel)
+, m_MessageHandler(mesgHandler)
+{
+}
+
+// -----------------------------------------------------------------------------
+FindBoundaryCells::~FindBoundaryCells() noexcept = default;
+
+// -----------------------------------------------------------------------------
+const std::atomic_bool& FindBoundaryCells::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> FindBoundaryCells::operator()()
+{
+  const ImageGeom imageGeometry = m_DataStructure.getDataRefAs<ImageGeom>(m_InputValues->ImageGeometryPath);
+  const SizeVec3 imageDimensions = imageGeometry.getDimensions();
+  const auto xPoints = static_cast<int64>(imageDimensions[0]);
+  const auto yPoints = static_cast<int64>(imageDimensions[1]);
+  const auto zPoints = static_cast<int64>(imageDimensions[2]);
+
+  auto featureIds = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeatureIdsArrayPath);
+  auto boundaryCells = m_DataStructure.getDataRefAs<Int8Array>(m_InputValues->BoundaryCellsArrayName);
+
+  int64 neighPoints[6] = {0, 0, 0, 0, 0, 0};
+  neighPoints[0] = -1 * (xPoints * yPoints);
+  neighPoints[1] = -1 * (xPoints);
+  neighPoints[2] = -1;
+  neighPoints[3] = 1;
+  neighPoints[4] = xPoints;
+  neighPoints[5] = xPoints * yPoints;
```

Comment:
```suggestion
  int64 neighPoints[6] = {(-1 * (xPoints * yPoints)), (-1 * (xPoints)), -1, 1, xPoints, (xPoints * yPoints)};
```
Just initialize it with the values in it no need for 6 extra copies

Also I would document what each represents:
Something like //{label1,label2,label3,label4,label5,label6} 
but this is an optional change